### PR TITLE
{quantum}[foss/2019a] Qiskit v0.12.0 w/ Python 3.7.2

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3109.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/38fb3582cc21b76631fbef9534e07a03 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2408.golett.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/50bff9d3288a6bb4a500b52b8647d72b for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2003.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/5874a13df3e0abff943ad7e8d224ff32 for a full test report.

File: easybuild/easyconfigs/q/Qiskit/Qiskit-0.12.0-foss-2019a-Python-3.7.2.eb [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+]
+
+postinstallcmds = [
+    # qiskit/providers/aer/VERSION.txt does not get installed, but is required by qiskit/providers/aer/version.py
+    "echo %s > %%(installdir)s/lib/python%%(pyshortver)s/site-packages/qiskit/providers/aer/VERSION.txt" % local_aerver,
+]
+
+sanity_check_paths = {
+    'files': [],
+    'dirs': ['lib/python%(pyshortver)s/site-packages'],
```
Comment (Micket) [inline easybuild/easyconfigs/q/Qiskit/Qiskit-0.12.0-foss-2019a-Python-3.7.2.eb]: So, what's the deal with sanity_check_paths and PythonBundle now again? I vaguely recall this being part of the block. Maybe?

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
vera-c1 - Linux centos linux 7.6.1810, Intel Xeon Processor (Skylake), Python 2.7.5
See https://gist.github.com/95fedc73290fb34acbe66bd36acd5a5e for a full test report.

File: easybuild/easyconfigs/q/Qiskit/Qiskit-0.12.0-foss-2019a-Python-3.7.2.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/q/Qiskit/Qiskit-0.12.0-foss-2019a-Python-3.7.2.eb]: Yes, indeed, this doesn't need to be here, it's exactly the same as the default. So, removing...

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
generoso-1 - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/715274fcddf160e3dd2555e6f3504198 for a full test report.

**Comment by Micket** [comment]
It seems to import the module anyway, but, Qiskit is doing strange things for me.
```
>>> import qiskit
/apps/Vera/software/MPI/GCC/8.2.0-2.31.1/OpenMPI/3.1.3/Qiskit/0.12.0-Python-3.7.2/lib/python3.7/site-packages/qiskit/version.py:116: UserWarning: Command ['/apps/Vera/software/Compiler/GCCcore/8.2.0/Python/3.7.2/bin/python3', '-m', 'pip', 'freeze'] exited with code 127: /apps/Vera/software/Compiler/GCCcore/8.2.0/Python/3.7.2/bin/python3: error while loading shared libraries: libpython3.7m.so.1.0: cannot open shared object file: No such file or directory
```

**Comment by boegel** [comment]
@Micket The `import qiskit` works (like it does in the sanity check) with exit code 0, but it's actually non-functional? 🤦‍♂

**Comment by boegel** [comment]
@Micket I'm seeing none of that, how come you are?

```
$ module load Qiskit/0.11.1-foss-2019a-Python-3.7.2
$ python
Python 3.7.2 (default, Aug 28 2019, 10:08:37)
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import qiskit
>>>
```

I do admit that running `pip freeze` when importing the Qiskit module is pretty horrible...

They've made some changes recently to make it more robust, see https://github.com/Qiskit/qiskit-terra/pull/2661/files

**Comment by Micket** [comment]
@boegel Are you possibly using rpath? Or have a libpython3.7m.so in system path?

**Comment by boegel** [comment]
Ah, hold on, that was for a different version... I am seeing the warning too:

```
$ ml Qiskit/0.12.0-foss-2019a-Python-3.7.2
$ python -c 'import qiskit'
/software/Qiskit/0.12.0-foss-2019a-Python-3.7.2/lib/python3.7/site-packages/qiskit/version.py:116: UserWarning: Command ['/user/gent/400/vsc40023/eb_scratch/CO7/skylake-ib/software/Python/3.7.2-GCCcore-8.2.0/bin/python', '-m', 'pip', 'freeze'] exited with code 127: /user/gent/400/vsc40023/eb_scratch/CO7/skylake-ib/software/Python/3.7.2-GCCcore-8.2.0/bin/python: error while loading shared libraries: libpython3.7m.so.1.0: cannot open shared object file: No such file or directory
  warnings.warn(str(exc))
```

**Comment by boegel** [comment]
This patch fixes the issue, seems reasonable to include that...

```diff
--- qiskit/version.py.orig 2019-09-15 18:24:58.754861000 +0200
+++ qiskit/version.py      2019-09-15 18:25:09.594520000 +0200
@@ -28,7 +28,7 @@
 def _minimal_ext_cmd(cmd):
     # construct minimal environment
     env = {}
-    for k in ['SYSTEMROOT', 'PATH']:
+    for k in ['SYSTEMROOT', 'PATH', 'LD_LIBRARY_PATH']:
         v = os.environ.get(k)
         if v is not None:
             env[k] = v

**Comment by boegel** [comment]
@Micket Patch added, no more warning now! :+1:

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3119.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/aadf5b1b7e72111a042f900ee42f4023 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2448.golett.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/691ab2aa8d0060301e896d21747192f5 for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 8 out of 8 (1 easyconfigs in this PR)
hebbe-s1.int.private - Linux centos 6.10, Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz, Python 2.7.14
See https://gist.github.com/41db91820091bb81d005d919c64781fb for a full test report.

**Comment by Micket** [comment]
Going in, thanks @boegel!

**Pull request merged**